assignments/this.js

```diff
@@ -13,14 +16,34 @@
 
 // code example for Window Binding
 
+
+
 // Principle 2
 
 // code example for Implicit Binding
 
+let MyInfo = function() {
+    this.name = 'Penny',
+    this.age = 'noneya',
+    this.daughter = 'Hope'
+};
+
+MyInfo.prototype.giveInfo = function() {
+    console.log(`${this.name}'s age is ${this.age} and she has a daughter named ${this.daughter}!`)
+}
+
+let info = new MyInfo();
+
+info.giveInfo()
+
 // Principle 3
 
 // code example for New Binding
```

Comment:
const Chessplayer = function(name, whiteOpening, blackOpening) {
    this.name = name;
    this.whiteOpening = whiteOpening;
    this.blackOpening = blackOpening;
    this.speak = function() {
      console.log(`Hello, my name is ${this.name} and my favorite openings are the ${this.whiteOpening} and the ${this.blackOpening}`);
    }
};
// when a new object is constructed, or invoked, as below, the 'this' keyword inside the function above is bound to the new object
const challenger = new Chessplayer('Fabiano Caruana', 'Queens Gambit', 'French Defense'); // here 'this' points to the new object, in this case the challenger